src/NuGet.Core/NuGet.Packaging/TopologicalSortUtility.cs

```diff
@@ -38,19 +38,25 @@ public static class TopologicalSortUtility
 
         private static int GetParentCount(List<PackageDependencyInfo> packages, string id)
         {
-            int count = 0;
+            var parentCount = 0;
+            var count = packages.Count;
 
-            foreach (var package in packages)
+            for (var i = 0; i < count; i++)
             {
-                if (package.Dependencies != null
-                    && package.Dependencies.Any(dependency =>
-                        string.Equals(id, dependency.Id, StringComparison.OrdinalIgnoreCase)))
+                var deps = packages[i].Dependencies;
+                var dependencies = deps as PackageDependency[] ?? deps.ToArray();
```

Comment:
💭 This is pretty dirty / prone to regress. If we use this approach for a short-term resolution we should still follow-up with a proper solution (modifying the `Dependencies` property to return a type with a struct enumerator and/or adding a `PackageDependencyInfo.ContainsDependency` method).